apps/OpenSpace/ext/launcher/src/profile/assetsdialog.cpp

```diff
@@ -259,3 +280,57 @@ void AssetsDialog::parseSelections() {
 void AssetsDialog::selected(const QModelIndex&) {
     _summary->setText(createTextSummary());
 }
+
+void AssetsDialog::searchTextChanged(const QString& text) {
+    if (!_assetProxyModel) {
+        return;
+    }
+    if (text.isEmpty()) {
+        setViewToBaseModel();
+        traverseToExpandSelectedItems(
+            *_assetTree,
+            _assetTreeModel,
+            _assetTreeModel.rowCount(_assetTreeModel.index(-1, 0)),
+            _assetTreeModel.index(-1, 0)
+        );
+    }
+    else {
+        _assetTree->setModel(_searchProxyModel);
+        _searchProxyModel->setFilterRegularExpression(text);
+        _assetTree->expandAll();
+    }
+}
+
+void SearchProxyModel::setFilterRegularExpression(const QString& pattern) {
+    _regExPattern = new QRegularExpression(
+        pattern,
+        QRegularExpression::CaseInsensitiveOption
+    );
+    QSortFilterProxyModel::setFilterRegularExpression(*_regExPattern);
+}
+
+bool SearchProxyModel::filterAcceptsRow(int sourceRow,
+                                        const QModelIndex& sourceParent) const
+{
+    QModelIndex idx = sourceModel()->index(sourceRow, 0, sourceParent);
+    return acceptIndex(idx);
+}
+
+bool SearchProxyModel::acceptIndex(const QModelIndex& idx) const {
+    if (idx.isValid()) {
+        QString text = idx.data(Qt::DisplayRole).toString();
+        if (!_regExPattern) {
+            return false;
+        }
+        QRegularExpressionMatchIterator matchIterator = _regExPattern->globalMatch(text);
+        if (matchIterator.hasNext()) {
+            return true;
+        }
+        for (int row = 0; row < idx.model()->rowCount(idx); ++row) {
+            if (acceptIndex(idx.model()->index(row, 0, idx))) {
+                return true;
+            }
+        }
+    }
+    return false;
```

Comment:
```suggestion
    if (!idx.isValid() || !_regExPattern) {
        return false;
    }
    QString text = idx.data(Qt::DisplayRole).toString();
    QRegularExpressionMatchIterator matchIterator = _regExPattern->globalMatch(text);
    if (matchIterator.hasNext()) {
        return true;
    }
    for (int row = 0; row < idx.model()->rowCount(idx); ++row) {
        if (acceptIndex(idx.model()->index(row, 0, idx))) {
            return true;
        }
    }

    return false;
```

Would reduce overall indentation

Reply:
All changes made